### ISSUE TITLE
Rule Change: Update no-shadow-restricted-names to report globalThis by default

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#19673]
### What rule do you want to change?

no-shadow-restricted-names 

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint no-shadow-restricted-names: "error"*/

const globalThis = {};

function test(globalThis) {}
```

### What does the rule currently do for this code?

No warnings reported

### What will the rule do after it's changed?

Report shadowing of `globalThis`.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Related: https://github.com/eslint/eslint/issues/19658

**Comment by sam3k** [comment]
TSC group has confirmed this for `v10`

**Comment by sethamus** [comment]
I’d be happy to work on this if you’re not planning to, @nzakas.

**Comment by nzakas** [comment]
@sethamus all yours. :+1:
